drivers/builtin/src/pk.c

```diff
@@ -1087,7 +1082,89 @@ int mbedtls_pk_verify_new(mbedtls_pk_type_t type, mbedtls_pk_context *ctx,
                           mbedtls_md_type_t md_alg, const unsigned char *hash,
                           size_t hash_len, const unsigned char *sig, size_t sig_len)
 {
-    return mbedtls_pk_verify_ext(type, NULL, ctx, md_alg, hash, hash_len, sig, sig_len);
+    if ((md_alg != MBEDTLS_MD_NONE || hash_len != 0) && hash == NULL) {
```

Comment:
What's the goal of re-implementing `mbedtls_pk_verify_ext` instead of just calling it as before?